src/logger/fileWriter.c

```diff
@@ -34,47 +35,55 @@
 #define WRITE_SUCCESS  0
 #define WRITE_FAIL     EOF
 
-typedef struct _FileBuffer {
-    char buffer[FILE_BUFFER_SIZE];
-    size_t index;
-} FileBuffer;
-
 static FIL *g_logfile;
 static xQueueHandle g_LoggerMessage_queue;
-static FileBuffer fileBuffer = {"", 0};
-
+static struct ring_buff file_buff;
 
-static void clear_file_buffer() {
-        fileBuffer.index = 0;
-        fileBuffer.buffer[0] = '\0';
+static void error_led(bool on)
+{
+        on ? LED_enable(3) : LED_disable(3);
 }
 
-static int writeFileBuffer()
+static FRESULT flush_file_buffer()
 {
-        unsigned int bw;
-        const int rc = f_write(g_logfile, fileBuffer.buffer,
-                               fileBuffer.index, &bw);
-        clear_file_buffer();
-        return rc;
+        FRESULT res = FR_OK;
+        char tmp[32];
+
+        pr_info("Flushing file buffer\r\n");
+        size_t chars = get_used(&file_buff);
+        while(0 < chars) {
+                if (chars > sizeof(tmp))
+                        chars = sizeof(tmp);
+
+                pr_info_int_msg("Chars is ", chars);
+                get_data(&file_buff, &tmp, chars);
+                if (g_logfile->fs) {
+                        unsigned int bw;
+                        res = f_write(g_logfile, &tmp, chars, &bw);
+                        error_led(FR_OK != res);
+                }
+
+                chars = get_used(&file_buff);
+        }
+
+        pr_info("Flushing file buffer DONE\r\n");
+        return res;
 }
 
-static void appendFileBuffer(const char * data)
+static FRESULT append_file_buffer(const char *data)
 {
-    size_t index = fileBuffer.index;
-    char * buffer = fileBuffer.buffer + index;
-
-    while(*data) {
-        *buffer++ = *data++;
-        index++;
-        if (index >= FILE_BUFFER_SIZE) {
-            *buffer = '\0';
-            writeFileBuffer();
-            index = fileBuffer.index;
-            buffer = fileBuffer.buffer + index;
+        FRESULT res = FR_OK;
+
+        const size_t size = strlen(data);
```

Comment:
think there could be a compliment to ring buffer's put_data that is called put_str that would avoid having to use strlen(), taking advantage of the assumption that strings are null terminated? it could avoid repeated called to strlen() in that case.